clients/src/main/resources/common/message/ElectPreferredLeadersRequest.json

```diff
@@ -0,0 +1,33 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{
+  "apiKey": 43,
+  "type": "request",
+  "name": "ElectPreferredLeadersRequest",
+  "validVersions": "0",
+  "fields": [
+    { "name": "TopicPartitions", "type": "[]TopicPartitions", "versions": "0+", "nullableVersions": "0+",
+      "about": "The topic partitions to elect the preferred leader of.",
+      "fields": [
+        { "name": "Topic", "type": "string", "versions": "0+",
+          "about": "The name of a topic." },
```

Comment:
@tombentley, you are correct that making this a mapKey would not inherently enforce uniqueness.  However, it would make it easier to add that enforcement to the code if you need to.  You could easily look up "are there any other topic entries that have the same name as this one, and set an error if so.